### PR TITLE
19929: use plugin lang if local role is created by plugin

### PR DESCRIPTION
**Pull request by shecken**
This attempts to fix [#19929](https://www.ilias.de/mantis/view.php?id=19929).

### CODE REVIEW

File: Services/AccessControl/classes/class.ilObjRole.php [hunk 1]
```diff
@@ -508,24 +508,28 @@ function getCountMembers()
...
+		if (preg_match("/^il_./", $test_str))
+		{
+			return $lng->txt($test_str);
 		}
 		
 		return $a_role_title;
 	}
 	
-	
+	public static function _removeObjectId($a_role_title) {
```
Comment (lechimp-p) [inline Services/AccessControl/classes/class.ilObjRole.php:522]: `private` plz.

File: Services/AccessControl/classes/class.ilObjRole.php [hunk 2]
```diff
@@ -508,24 +508,28 @@ function getCountMembers()
...
+		$test_str = explode('_',$a_role_title);
```
Comment (lechimp-p) [inline Services/AccessControl/classes/class.ilObjRole.php]: Why `$test_str`? To me this name is misleading. Nice refactoring, though.

**Review by lechimp-p — CHANGES_REQUESTED** [review]
LGTM in general, minor renaming, thx for refactoring.

File: Services/AccessControl/classes/class.ilObjRole.php (hunk 1 above)
Comment (shecken) [reply Services/AccessControl/classes/class.ilObjRole.php:522]: I need this public because of use in ilObjectRolePermissionTableGUI.php line 685.

**Comment by chfsx** [comment]
@smeyer-ilias could you please have a look at this PR? Thanks

**Comment by klees** [comment]
Hi @smeyer-ilias,
any chance that this will be merged some time?
Regards!

**Comment by chfsx** [comment]
TB: We will discuss this PR during next JF

**Comment by smeyer-ilias** [comment]
Thanks, for providing a bugfix.

**Pull request merged**